### PR TITLE
feat(settings): skip calling graphql hooks if user does not have the right permissions

### PR DESCRIPTION
**Pull request by aditya-radhakrishnan**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/identity/user/ViewInviteTokenModal.tsx [hunk 1]
```diff
@@ -79,7 +82,15 @@ export default function ViewInviteTokenModal({ visible, onClose }: Props) {
                 <ModalSectionParagraph>
                     Generate a new invite link! Note, any old links will <b>cease to be active</b>.
                 </ModalSectionParagraph>
-                <CreateInviteTokenButton onClick={() => createNativeUserInviteToken({})} size="small" type="text">
+                <CreateInviteTokenButton
+                    onClick={() => {
+                        if (canManageUserCredentials) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/identity/user/ViewInviteTokenModal.tsx:87]: Do we need this check?

File: datahub-web-react/src/app/settings/AccessTokens.tsx [hunk 2]
```diff
@@ -98,6 +98,7 @@ export const AccessTokens = () => {
         data: tokensData,
         refetch: tokensRefetch,
     } = useListAccessTokensQuery({
+        skip: !canGeneratePersonalAccessTokens,
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/settings/AccessTokens.tsx:101]: Amazing!

File: datahub-web-react/src/app/identity/user/ViewInviteTokenModal.tsx (hunk 1 above)
Comment (aditya-radhakrishnan) [reply datahub-web-react/src/app/identity/user/ViewInviteTokenModal.tsx:87]: I don't think so but I also feel like it doesn't hurt?

File: datahub-web-react/src/app/identity/user/UserList.tsx [hunk 3]
```diff
@@ -133,7 +133,11 @@ export const UserList = () => {
                         showSizeChanger={false}
                     />
                 </UserPaginationContainer>
-                <ViewInviteTokenModal visible={isViewingInviteToken} onClose={() => setIsViewingInviteToken(false)} />
+                <ViewInviteTokenModal
```
[the remark this answers is not in the document]
Comment (aditya-radhakrishnan) [reply datahub-web-react/src/app/identity/user/UserList.tsx]: Agreed! Fixed

File: datahub-web-react/src/app/identity/user/ViewInviteTokenModal.tsx (hunk 1 above)
Comment (aditya-radhakrishnan) [reply datahub-web-react/src/app/identity/user/ViewInviteTokenModal.tsx]: Removed

**Pull request merged**